### PR TITLE
fixes #605 : Added Layout Test and Options Menu Test for MainMenuActivity

### LINKED ISSUE #605: Add Test for Layout and Options Menu in MainActivityTest
*MainActivityTest* implemented by @yadav-rahul doesn't have test for checking layout and options menu.
As Options Menu are important component in ODK-collect , there must be a local unit test to check whether intents from options menu are working properly or not.

- [ ] Add test for main activity layout.
- [ ] Add test for each options menu.

### PR DESCRIPTION
**Pull request by lokeshkvn**
fixes #605 

Added Layout Test and Options Menu Test for *MainMenuActivity*

- [x] Add test for main activity layout.
- [x] Add test for each options menu.

### CODE REVIEW

File: collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java [hunk 1]
```diff
@@ -52,6 +64,36 @@ public void titleTest() throws Exception {
...
+        mainMenuActivity.onOptionsItemSelected(menu.getItem(1));
+        ShadowActivity shadowGeneralPreferencesActivity = shadowOf(mainMenuActivity);
+        Intent startedIntentGeneralPreferences = shadowGeneralPreferencesActivity.getNextStartedActivity();
+        ShadowIntent shadowIntentGeneralPreferences = shadowOf(startedIntentGeneralPreferences);
+        assertEquals(PreferencesActivity.class.getName(),shadowIntentGeneralPreferences.getIntentClass().getName());
+
+        //Test for AdminPreferencesActivity
+        String menuTitle = mainMenuActivity.getResources().getString(R.string.admin_preferences);
+        String shadowTitle = menu.getItem(2).getTitle().toString();
+        assertEquals(shadowTitle,menuTitle);
```
Comment (yadav-rahul) [inline collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java]: formatting needed here.

**Comment by yadav-rahul** [comment]
@lokeshkvn Why making `shadow` instance of `MainMenuActivity` twice( [here](https://github.com/opendatakit/collect/pull/609/files#diff-0cd9f80ca7a89f28f16c6e01a6a05755R78) and [here](https://github.com/opendatakit/collect/pull/609/files#diff-0cd9f80ca7a89f28f16c6e01a6a05755R85) )? I think it can easily be done by making a single variable. Also, try same for `startedIntent` and `ShadowIntent`. :)

File: collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java [hunk 2]
```diff
@@ -52,6 +64,36 @@ public void titleTest() throws Exception {
...
+        assertEquals(AboutPreferencesActivity.class.getName(),shadowIntentAbout.getIntentClass().getName());
+
+        //Test for PreferencesActivity
+        mainMenuActivity.onOptionsItemSelected(menu.getItem(1));
+        ShadowActivity shadowGeneralPreferencesActivity = shadowOf(mainMenuActivity);
+        Intent startedIntentGeneralPreferences = shadowGeneralPreferencesActivity.getNextStartedActivity();
+        ShadowIntent shadowIntentGeneralPreferences = shadowOf(startedIntentGeneralPreferences);
+        assertEquals(PreferencesActivity.class.getName(),shadowIntentGeneralPreferences.getIntentClass().getName());
+
+        //Test for AdminPreferencesActivity
```
Comment (yadav-rahul) [inline collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java]: Is test for `AdminPreferencesActivity` complete?

Comment (lokeshkvn) [reply collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java]: While testing intent of AdminPreferencesActivity as it is using shared preference  of admin password  in MENU_ADMIN , it is giving null pointer exception. I tried edit() sharedpreferences in setup but then also it is giving NPE. so I written test to just assert title of menu not for intents.  
Do you have any other plans for this ?

Comment (yadav-rahul) [reply collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java]: I see. This one would be a piece of cake for `Espresso`, but for now, I think asserting `title` is fine.

**Comment by yadav-rahul** [comment]
@lokeshkvn Code looks good :+1: . Just few more things to do.

1. First commit has a small type error in its commit message. Please fix it.
2. Squash both commits to one as second commit is not adding anything new. 
3. Also, try to add default formatting using `Ctrl` + `Alt` + `Shift` + `L` for Ubuntu or `Option` + `Command` + `L` for mac.

File: collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java (hunk 2 above)
Comment (lokeshkvn) [reply collect_app/src/test/java/org/odk/collect/android/activities/MainActivityTest.java]: Yes, this should be done using ```Espresso``` 😃

**Comment by lokeshkvn** [comment]
@lognaturel @yadav-rahul done with changes 👍

**Comment by yadav-rahul** [comment]
@lognaturel Looks good to me now.

**Pull request merged**